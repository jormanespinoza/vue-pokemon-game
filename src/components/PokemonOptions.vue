<template>
  <div class="options-container">
    <ul>
      <li
        v-for="{ id, name, answer = '' } in pokemons"
        :key="id"
        :class="answer"
        @click="$emit('pokemonSelected', id)"
      >
        {{ name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true
    }
  }
}
</script>

<style scope>
/* Pokemon Options */
.options-container {
  display: flex;
  justify-content: center;
}

ul {
  list-style-type: none;
  padding: 0;
}

li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

li.correct {
  background-color: #2d649b;
  color: #fff;
}

li.wrong {
  background-color: #da4f4f;
  color: #fff;
}
</style>