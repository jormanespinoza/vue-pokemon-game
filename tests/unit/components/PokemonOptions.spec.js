import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'

import { mockedPokemos } from '../mocks/pokemons.mock'

describe('PokemonOptions.vue', () => {
    let wrapper
    let pokemonOptions

    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: mockedPokemos
            }
        })

        pokemonOptions = wrapper.findAll('li')
    })

    test('should match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
        // expect(wrapper.html()).toMatchInlineSnapshot(`
        //   <div class='options-container'>
        //     <ul>
        //       <li class=''>Bulbasaur</li>
        //       <li class=''>Ivysaur</li>
        //       <li class=''>Venusaur</li>
        //       <li class=''>Charmander</li>
        //     </ul>
        //   </div>
        // `)
    })

    test('should show 4 pokemon options', async () => {
        expect(pokemonOptions.length).toBe(4)
        mockedPokemos.forEach((pokemon, index) => expect(pokemon.name).toBe(pokemonOptions[index].text()))
    })

    test('should $emit event on pokemon option click', () => {
        pokemonOptions.forEach((pokemonOption, index) => {
            pokemonOption.trigger('click')

            expect(wrapper.emitted('pokemonSelected').length).toBe(index + 1)
            expect(wrapper.emitted('pokemonSelected')[index]).toEqual([mockedPokemos[index].id])
        })
    })
})
