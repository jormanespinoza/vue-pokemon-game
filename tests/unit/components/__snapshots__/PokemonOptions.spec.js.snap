// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions.vue should match snapshot 1`] = `
<div class="options-container">
  <ul>
    <li class="">Bulbasaur</li>
    <li class="">Ivysaur</li>
    <li class="">Venusaur</li>
    <li class="">Charmander</li>
  </ul>
</div>
`;
